py4cast/lightning.py

```diff
@@ -180,19 +200,67 @@ class AutoRegressiveLightning(pl.LightningModule):
     Auto-regressive lightning module for predicting meteorological fields.
     """
 
-    def __init__(self, hparams: ArLightningHyperParam, *args, **kwargs):
+    def __init__(
+        self,
+        dataset_name: str,
+        dataset_conf: str,
+        model_name: str,
+        model_conf: Union[str, None],
+        batch_size: int,
+        num_input_steps: int,
+        num_pred_steps_train: int,
+        num_pred_steps_val_test: int,
+        num_inter_steps: int,
+        lr: float,
+        loss: str,
+        training_strategy: str,
+        use_lr_scheduler: bool,
+        precision: str,
+        no_log: bool,  # doublon avec trainer.logger
+        channels_last: bool,
+        len_train_loader: int,
+        dataset_info,
+        save_path: str,
+        *args,
+        **kwargs,
+    ):
         super().__init__(*args, **kwargs)
 
-        self.save_hyperparameters()  # write hparams.yaml in save folder
+        # Gather hyperparameters in a class
+        hparams = ArLightningHyperParam(
```

Comment:
`self.hparams = ... ` non ?

Reply:
After I instantiate the class ArLightningHyperParam, the script calls self.save_hyperparameters(). It creates an instance of self.hparams, so it would be overwriten. But either way, the both (hparams and self.hparams) are very similar. 
You think I should overwrite it ?